vm/devices/storage/scsi_buffers/src/lib.rs

```diff
@@ -204,6 +205,52 @@ impl LockedRange for LockedIoVecs {
     }
 }
 
+/// An implementation of [`MemoryWrite`] that provides semantically
+/// correct results. Specifically, it always returns a `ReadOnly` error
+/// when attempting to write to it.
+pub struct PermissionedMemoryWriter<'a> {
```

Comment:
This doesn't need to be `pub` since it's only exposed via `impl MemoryWrite` elsewhere.